packages/@aws-cdk/aws-kinesisfirehose-alpha/lib/encryption.ts

```diff
@@ -0,0 +1,44 @@
+import { StreamEncryptionType } from './delivery-stream';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+
+/**
+ * Represents server-side encryption for a Kinesis Firehose Delivery Stream.
+ */
+export abstract class StreamEncryption {
+  /**
+   * No server-side encryption is configured.
+   */
+  public static unencrypted(): StreamEncryption {
+    return new (class extends StreamEncryption {
+    }) (StreamEncryptionType.UNENCRYPTED);
+  }
+
+  /**
+   * Configure server-side encryption using an AWS owned key.
+   */
+  public static awsOwnedKey(): StreamEncryption {
+    return new (class extends StreamEncryption {
+    }) (StreamEncryptionType.AWS_OWNED);
+  }
+
+  /**
+   * Configure server-side encryption using customer managed keys.
+   *
+   * @param encryptionKey the KMS key for the delivery stream.
+   */
+  public static customerManagedKey(encryptionKey?: IKey): StreamEncryption {
+    return new (class extends StreamEncryption {
+
+    }) (StreamEncryptionType.CUSTOMER_MANAGED, encryptionKey);
+  }
```

Comment:
Why not something like this?
```
export interface StreamEncryption {
  encryptionKey?: string;
  type: StreamEncryptionType
}

/**
 * Options for server-side encryption of a delivery stream.
 */
export enum StreamEncryptionType {
  /**
   * Data in the stream is stored unencrypted.
   */
  UNENCRYPTED,

  /**
   * Data in the stream is stored encrypted by a KMS key managed by the customer.
   */
  CUSTOMER_MANAGED,

  /**
   * Data in the stream is stored encrypted by a KMS key owned by AWS and managed for use in multiple AWS accounts.
   */
  AWS_OWNED,
}
```

Reply:
That would also work but is slightly less preferred because of it's nested nature (although at one level of nesting it's still fine). Using this class design follows the existing pattern in [TableV2's encryption](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-dynamodb/lib/encryption.ts) and also opens up the option to add private methods to set additional values in the future.